In [1]:
"""check if file exists in the current working directory
import os

print("current working directory: ", os.getcwd())
data_file = '\SQUIDParams\\20230310_002446SQUIDParams.pkl'
Absolute_path = os.getcwd()+data_file
print("Absolute path: ", Absolute_path)
print("Directory: ", os.path.dirname(Absolute_path))
print(os.path.isfile(Absolute_path))
"""
print("code for check the file path")

code for check the file path


In [26]:
import pickle
import gzip

data_file = 'results_1717115674_c8.pklz'

with gzip.open(data_file, "rb") as file:
    data = pickle.load(file)

print(data.keys())

dict_keys(['tune_ctime', 8])


In [5]:
for key1 in data.keys():
    
    if type(data[key1]) == dict:
        for key2 in data[key1].keys():
            print(f"{key2}: {data[key1][key2]}")
            #here have 29 keys in data[8]
    
    else:    
        print(f"{key1}: {data[key1]}")
        
    

tune_ctime: 1717115674
sa_bias_dac_to_uA: 0.0025223473681034937
sa_bias_dac: 19307
sa_bias_uA: 48.69896063597415
sa_fb_dac_to_uA: 0.007165634656294612
sa_fb_dac: [    0    10    20 ... 19970 19980 19990]
sa_fb_uA: [0.00000000e+00 4.83918899e-02 9.67837798e-02 ... 9.66386042e+01
 9.66869961e+01 9.67353879e+01]
sa_out_m_adu2mV: 0.0007075
sa_offset_dac_to_mV: 0.0011595249141099942
sa_out_b_adu2mV: 10.745317379057315
sa_out_adu: [-1770.10909091 -1860.90606061 -1952.48181818 ... -3723.97878788
 -3800.04242424 -3878.80606061]
sa_out_mV: [9.4929652  9.42872634 9.36393649 ... 8.11060239 8.05678736 8.00106209]
mce_sa_fb_lock_dac: 1106
mce_sa_fb_lock_uA: 7.925191929861841
mce_sa_out_lock_mV: 7.920741116936103
sa_phi0_fb_uA: 35.398235202095385
M_ssa_fb_pH: 58.416297767228684
sa_out_max_idx: 311
sa_out_max_upslope_idx: 131
sa_out_max_downslope_idx: 11
Vmod_adu: 11838.348484848484
Vmod_mV: 8.375631553030303
dV_ADU_dSAFB_DAC_downslope: -10.338484848484823
dV_nV_dSAFB_uphi0_downslope: -36.13352146969

In [17]:
uA = data[8]['sa_bias_uA']
print(uA)
dac = data[8]['sa_bias_dac']
dac_to_uA = data[8]['sa_bias_dac_to_uA']
print(f"{dac} * {dac_to_uA} = ",dac*dac_to_uA)

48.69896063597415
19307 * 0.0025223473681034937 =  48.69896063597415


In [19]:
sa_fb_dac = data[8]['sa_fb_dac']
# for i in sa_fb_dac:
#     print(i)
print(sa_fb_dac[:10],max(sa_fb_dac))
print(len(sa_fb_dac)) # 2000 why?

#print(24*60*60/2000) # 43.2s/point?

[ 0 10 20 30 40 50 60 70 80 90] 19990
2000


In [23]:
sa_out_adu = data[8]['sa_out_adu']
print(len(sa_out_adu)) # 2000
sa_out_m_adu2mV = data[8]['sa_out_m_adu2mV']
sa_out_b_adu2mV = data[8]['sa_out_b_adu2mV']
sa_out_mV = data[8]['sa_out_mV']
print(sa_out_mV[:10],max(sa_out_mV))

sa_out_mV = sa_out_adu * sa_out_m_adu2mV + sa_out_b_adu2mV
print(sa_out_mV[:10],max(sa_out_mV))


2000
[9.4929652  9.42872634 9.36393649 9.2964753  9.23962445 9.17088976
 9.10728122 9.04796914 8.9767732  8.91556802] 15.534095447239133
[9.4929652  9.42872634 9.36393649 9.2964753  9.23962445 9.17088976
 9.10728122 9.04796914 8.9767732  8.91556802] 15.534095447239133
19.120948932087618


In [ ]:
dV_nV_dSAIN_pA_downslope = data[8]['dV_nV_dSAIN_pA_downslope']


tune_ctime  recode time?  

(
DAC-->look like a unit?
mce   (mce data)?
ADU  ?       
sa-->  “SQUID series arrays” (SSAs);     
M--> mutual inductance (pH is Pico Henry = 10^-12 Henry)      
fb --> feedback
uA-->microampere (µA) mV-->millivolt (mV) .      
)  

`*`sa_bias_dac_to_uA * sa_bias_dac = sa_bias_uA --> 19307 * 0.0025223473681034937 ~=  48.7    

sa_fb_dac_to_uA * sa_fb_dac = `*`sa_fb_uA ('SSA Feedback Current (uA)' is xaxis)   
sa_fb_dac & sa_fb_uA: array length is 2000    

sa_out_mV = sa_out_adu * sa_out_m_adu2mV + sa_out_b_adu2mV       
`*`sa_out_mV (SSA Output Voltage (mV) is yaxis) array length is 2000

?mce_sa_fb_lock_dac: 1106
?mce_sa_fb_lock_uA: 7.925191929861841
?mce_sa_out_lock_mV: 7.920741116936103

`*`sa_phi0_fb_uA: -->35.4      

`*` M_ssa_fb_pH: 58.416297767228684  

?sa_out_max_idx: 311
?sa_out_max_upslope_idx: 131
?sa_out_max_downslope_idx: 11

Vmod_adu: 11838.348484848484           
`*` Vmod_mV = ? --> 8.376

`*`dV_ADU_dSAFB_DAC_downslope  --> -10.34                        
`*`dV_nV_dSAFB_uphi0_downslope   --> -36.13                       

dV_nV_dSAIN_pA_downslope = ? --> -1.5115090654656795             

`*`dV_ADU_dSAFB_DAC_upslope  -->9.82  
`*`dV_nV_dSAFB_uphi0_upslope   -->34.31

dV_nV_dSAIN_pA_upslope: 1.435173803871819     

`*`dV_ADU_dSAFB_DAC_mceslope: 6.888787878787859                
`*`dV_nV_dSAFB_uphi0_mceslope: 24.07665807575751   
            
dV_nV_dSAIN_pA_mceslope: 1.0071558339019047                  

`*` is in figure ...4.37.24 PM